src/megolm/ratchet.rs

```diff
@@ -199,38 +245,10 @@ impl Ratchet {
     }
 }
 
-#[derive(Serialize, Deserialize, Zeroize)]
-#[zeroize(drop)]
-pub(super) struct RatchetPickle {
-    key: Vec<u8>,
-    counter: u32,
-}
-
-impl From<Ratchet> for RatchetPickle {
-    fn from(ratchet: Ratchet) -> Self {
-        Self { key: ratchet.inner.into(), counter: ratchet.counter }
-    }
-}
-
-impl TryFrom<RatchetPickle> for Ratchet {
-    type Error = MegolmRatchetUnpicklingError;
-
-    fn try_from(pickle: RatchetPickle) -> Result<Self, Self::Error> {
-        Ok(Ratchet {
-            inner: pickle
-                .key
-                .clone()
-                .try_into()
-                .map_err(|_| MegolmRatchetUnpicklingError::InvalidKeyLength(pickle.key.len()))?,
-            counter: pickle.counter,
-        })
-    }
-}
-
 #[derive(Error, Debug)]
-pub enum MegolmRatchetUnpicklingError {
-    #[error("Invalid Megolm ratchet key length: expected 128, got {0}")]
-    InvalidKeyLength(usize),
+pub enum RatchetError {
```

Comment:
Do we expect to ever use this outside deserialization? If not, can we please name it something like `RatchetDeserializationError`?

Reply:
But it's not a deserialization error. It is used in the deserialization implementation and converted into a `serde::de::Error::custom`. I don't think this makes much sense and the type should anyways be private to this module.

Reply:
Did so.